### ISSUE TITLE
[Bug] Fenix thinks that its reader mode is insecure

### ISSUE THREAD

**Issue by linsui** [mozilla-mobile/fenix#16602]
[comment]: # (Please do your best to search for duplicate issues before filing a new issue so we can keep our issue board clean)
[comment]: # (Every issue should have the exact bug and steps to reproduce described in it. Please do not file feedback list tickets as it is difficult to parse them and address their individual points)
[comment]: # (Read https://github.com/mozilla-mobile/fenix#i-want-to-file-an-issue for more information)

## Steps to reproduce
Enable reader mode on an https site.
### Expected behavior
It should be shown as secure connection.
### Actual behavior
It's shown as insecure connection.
![IMG_20201117_004706](https://user-images.githubusercontent.com/36977733/99283167-87fb5680-282c-11eb-92c3-4353194723fe.jpg)

![IMG_20201117_004613](https://user-images.githubusercontent.com/36977733/99283180-8d58a100-282c-11eb-9266-f2b76e62859d.jpg)

![IMG_20201117_005023](https://user-images.githubusercontent.com/36977733/99283092-7154ff80-282c-11eb-801f-2846e55d0357.jpg)

### Device information

* Android device: Android 10
* Fenix version: Nightly 201114 17:01 (Build #2015775627)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-8807)

**Comment by csadilek** [comment]
Desktop shows the insecure lock icon too but provides an explanation and shows a different icon in the toolbar. We should consider doing this too:

![desktop-reader](https://user-images.githubusercontent.com/472523/99328113-6d83a600-2849-11eb-82d1-e004357aafff.png)

**Comment by linsui** [comment]
It's different on my desktop.
![sshot-015](https://user-images.githubusercontent.com/36977733/99346931-af364000-288d-11eb-85f2-6ead43769826.png)

**Comment by cadeyrn** [comment]
@csadilek Firefox on desktop no longer shows an insecure lock icon. The icon shown by @linsui is used since Firefox 79 (https://bugzilla.mozilla.org/show_bug.cgi?id=1570678).

**Comment by csadilek** [comment]
@cadeyrn thanks! I took the screenshot in yesterday's Nightly (Desktop), but not sure what happened there. Can't reproduce this now. Either way I agree we should fix this.

**Comment by codrut-topliceanu** [comment]
At the moment `EngineObserver.kt` receives data regarding the security of the site through:

 `override fun onSecurityChange(secure: Boolean, host: String?, issuer: String?)`

Wikipedia page in normal mode -> `secure = true`, `host = en.m.wikipedia.org`, `issuer = DigiCert Inc`

Wikipedia page in reader mode -> `secure = false`, `host = null`, `issuer = null`

So if UX decides we need to show reader mode as secure (or some other variation) if original site is secure than perhaps we need GV to provide some extra data ?

**Comment by lobontiumira** [comment]
Reproducible on RC 89.1.0 with Samsung Galaxy Tab A6 (Android 5.1.1).

**Comment by data-sync-user** [comment]
➤ Constantin Codrut Topliceanu commented:

🚩 Flag added

Waiting on UX

**Comment by codrut-topliceanu** [comment]
> ➤ Constantin Codrut Topliceanu commented:
> 
> 🚩 Flag added
> 
> Waiting on UX

You're a bit late to the party @data-sync-user , that's a 6 month old comment 😆 

Status: [bugzilla ticket](https://bugzilla.mozilla.org/show_bug.cgi?id=1686210) still open

**Comment by lobontiumira** [comment]
Reproducible on the 6/28 Nightly build, with Samsung Galaxy Note 8 (Android 9).

**Comment by lobontiumira** [comment]
Still reproducible on Nightly 94.0a1 from 9/24, Beta 93.0.0-beta.4, and RC 92.1.1 with OnePlus 5T (Android 10).

**Comment by sflorean** [comment]
This is still reproducible on Fenix versions.

**Comment by cadeyrn** [comment]
It's still reproducible.

**Comment by gabrielluong** [comment]
Moved to bugzilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1813872

Change performed by the [Move to Bugzilla add-on](https://addons.mozilla.org/en-US/firefox/addon/move-to-bugzilla/).
